src/test/storage/value_segment_test.cpp

```diff
@@ -9,42 +9,56 @@
 
 namespace opossum {
 
-// class StorageValueSegmentTest : public BaseTest {
-//  protected:
-//   ValueSegment<int> int_value_segment;
-//   ValueSegment<std::string> string_value_segment;
-//   ValueSegment<double> double_value_segment;
-// };
-
-// TEST_F(StorageValueSegmentTest, GetSize) {
-//   EXPECT_EQ(int_value_segment.size(), 0u);
-//   EXPECT_EQ(string_value_segment.size(), 0u);
-//   EXPECT_EQ(double_value_segment.size(), 0u);
-// }
-
-// TEST_F(StorageValueSegmentTest, AddValueOfSameType) {
-//   int_value_segment.append(3);
-//   EXPECT_EQ(int_value_segment.size(), 1u);
-
-//   string_value_segment.append("Hello");
-//   EXPECT_EQ(string_value_segment.size(), 1u);
-
-//   double_value_segment.append(3.14);
-//   EXPECT_EQ(double_value_segment.size(), 1u);
-// }
-
-// TEST_F(StorageValueSegmentTest, AddValueOfDifferentType) {
-//   int_value_segment.append(3.14);
-//   EXPECT_EQ(int_value_segment.size(), 1u);
-//   EXPECT_THROW(int_value_segment.append("Hi"), std::exception);
-
-//   string_value_segment.append(3);
-//   string_value_segment.append(4.44);
-//   EXPECT_EQ(string_value_segment.size(), 2u);
-
-//   double_value_segment.append(4);
-//   EXPECT_EQ(double_value_segment.size(), 1u);
-//   EXPECT_THROW(double_value_segment.append("Hi"), std::exception);
-// }
+class StorageValueSegmentTest : public BaseTest {
+ protected:
+  ValueSegment<int> int_value_segment;
+  ValueSegment<std::string> string_value_segment;
+  ValueSegment<double> double_value_segment;
+};
+
+TEST_F(StorageValueSegmentTest, GetSize) {
+  EXPECT_EQ(int_value_segment.size(), 0u);
+  EXPECT_EQ(string_value_segment.size(), 0u);
+  EXPECT_EQ(double_value_segment.size(), 0u);
+}
+
+TEST_F(StorageValueSegmentTest, AddValueOfSameType) {
+  int_value_segment.append(3);
+  EXPECT_EQ(int_value_segment.size(), 1u);
+
+  string_value_segment.append("Hello");
+  EXPECT_EQ(string_value_segment.size(), 1u);
+
+  double_value_segment.append(3.14);
+  EXPECT_EQ(double_value_segment.size(), 1u);
+}
+
+TEST_F(StorageValueSegmentTest, GetValues) {
+  int_value_segment.append(3);
+  EXPECT_EQ(int_value_segment.values().at(0), 3);
+  EXPECT_EQ(type_cast<int>(int_value_segment[ChunkOffset{0}]), 3);
+
+  string_value_segment.append("Hello");
+  EXPECT_EQ(string_value_segment.values().at(0), "Hello");
+  EXPECT_EQ(type_cast<std::string>(string_value_segment[ChunkOffset{0}]), "Hello");
+
+  double_value_segment.append(3.14);
+  EXPECT_EQ(double_value_segment.values().at(0), 3.14);
+  EXPECT_EQ(type_cast<double>(double_value_segment[ChunkOffset{0}]), 3.14);
+}
```

Comment:
Very nice testing for all three value_segments. 

Maybe check for invalid index access.